examples/moonsniff/post-processing.lua

```diff
@@ -0,0 +1,128 @@
+	parser:option("-n --nrbuckets", "Size of a bucket for the resulting histogram."):args(1):convert(tonumber):default(1)
+	parser:flag("-d --debug", "Create debug information. Instead of processing the input files normally, they are translated into human readable csv files.")
+	parser:flag("-p --profile", "Profile the application. May decrease the overall performance.")
+        return parser:parse()
+end
+
+
+--- Main entry of the program
+function master(args)
+	if args.input then INPUT_PATH = args.input end
+
+	-- determine the input type
+	if string.match(args.input, ".*%.pcap") then
+		MODE = MODE_PCAP
+	elseif string.match(args.input, ".*%.mscap") then
+		MODE = MODE_MSCAP
+	else
+		log:err("Input with unknown file extension.\nCan only process .pcap or .mscap files.")
+	end
+
+	print(MODE)
+	local PRE
+	local POST
+
+
+	-- determine which is the post and which is the pre file
+	if MODE == MODE_MSCAP then
+		if not args.second then log:fatal("Detected .mscap file but there was no second file. Single .mscap files cannot be processed.") end
+
+		if string.match(args.input, ".*%-pre%.mscap") and string.match(args.second, ".*%-post%.mscap") then
+			PRE = args.input
+			POST = args.second
+
+		elseif string.match(args.second, ".*%-pre%.mscap") and string.match(args.input, ".*%-post%.mscap") then
+			POST = args.input
+			PRE = args.second
+		else
+			log:fatal("Could not decide which file is pre and which post. Pre should end with -pre.mscap and post with -post.mscap.")
+		end
+	end
+
+	if MODE == MODE_PCAP then
+
+		if not args.second then log:fatal("Detected .pcap file but there was no second file. Single .pcap files cannot be processed.") end
+
+		if string.match(args.input, ".*%-pre%.pcap") and string.match(args.second, ".*%-post%.pcap") then
+			PRE = args.input
+			POST = args.second
+
+		elseif string.match(args.second, ".*%-pre%.pcap") and string.match(args.input, ".*%-post%.pcap") then
+			POST = args.input
+			PRE = args.second
+		else
+			log:fatal("Could not decide which file is pre and which post. Pre should end with -pre.mscap and post with -post.mscap.")
+		end
+	end
+
+
+	print(PRE)
+	print(POST)
+
+
+	-- retrieve file size in order to give an estimation for the processing speed
+	local file = assert(io.open(PRE, "r"))
+        local size = fsize(file)
+        file:close()
+        file = assert(io.open(POST, "r"))
+        size = size + fsize(file)
+        file:close()
+        log:info("File size: " .. size / 1e9 .. " [GB]")
+        local nClock = os.clock()
+        if args.profile then profile.start("-fl", "profile.log") end
+
+	local packets = 0
+
+	-- run correct matching file
+	if MODE == MODE_PCAP then
+		local tbb = require "tbbmatch"
+		packets = tbb.match(PRE, POST, args)
+	elseif MODE == MODE_MSCAP then
+		local arr = require "arrmatch"
+		packets = arr.match(PRE, POST, args)
+	end
+
+	-- finish operations
+	if args.profile then profile.stop() end
+
+        local elapsed = os.clock() - nClock
```

Comment:
avoid using os.clock (it's CPU time, not wall time), use the libmoon timers instead

Reply:
correction: libmoon.getTime() would require DPDK startup which is unnecessary in this case

using utils: getMonotonicTime() instead